### PR TITLE
makepkg 6.0.1

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

### CODE REVIEW

**Comment by chenrui333** [comment]
for 10.15 build
```
  meson.build:83:6: ERROR: Problem encountered: ngettext not found but NLS support requested
```

**Comment by danielnachun** [comment]
I was able to get the Linux build fixed.  I am still not sure how to fix the error with `ngettext` - it's not able to find `libintl`, possibly `gettext` does not have a `pkg-config` file.

**Comment by cho-m** [comment]
It is probably only looking for `glibc` (or `musl`) system copy rather than `gettext`:
```meson
if get_option('i18n')
  if not cc.has_function('ngettext')
    libintl = cc.find_library('intl', required : false, static: get_option('buildstatic'))
```

As is, I think we would need to disable `i18n` on macOS.

`meson` does have document finding intl at:
- https://mesonbuild.com/Dependencies.html#builtin
- https://mesonbuild.com/Dependencies.html#intl

**Comment by danielnachun** [comment]
```
../lib/libalpm/diskspace.c:165:30: error: no member named 'f_mntonname' in 'struct statvfs'
                  STRDUP(mp->mount_dir, fsp->f_mntonname, free(mp); RET_ERR(handle, ALPM_ERR_MEMORY, NULL));
                                        ~~~  ^
  ../lib/libalpm/util.h:58:38: note: expanded from macro 'STRDUP'
  #define STRDUP(r, s, action) do { if(s != NULL) { r = strdup(s); if(r == NULL) { _alpm_alloc_fail(strlen(s)); action; } } else { r = NULL; } } while(0)
                                       ^
  ../lib/libalpm/diskspace.c:165:30: error: no member named 'f_mntonname' in 'struct statvfs'
                  STRDUP(mp->mount_dir, fsp->f_mntonname, free(mp); RET_ERR(handle, ALPM_ERR_MEMORY, NULL));
                                        ~~~  ^
  ../lib/libalpm/util.h:58:62: note: expanded from macro 'STRDUP'
  #define STRDUP(r, s, action) do { if(s != NULL) { r = strdup(s); if(r == NULL) { _alpm_alloc_fail(strlen(s)); action; } } else { r = NULL; } } while(0)
                                                               ^
  ../lib/libalpm/diskspace.c:165:30: error: no member named 'f_mntonname' in 'struct statvfs'
                  STRDUP(mp->mount_dir, fsp->f_mntonname, free(mp); RET_ERR(handle, ALPM_ERR_MEMORY, NULL));
                                        ~~~  ^
  ../lib/libalpm/util.h:58:106: note: expanded from macro 'STRDUP'
  #define STRDUP(r, s, action) do { if(s != NULL) { r = strdup(s); if(r == NULL) { _alpm_alloc_fail(strlen(s)); action; } } else { r = NULL; } } while(0)
```

I was able to get past the configuration issues but now we have this on macOS.  I think the problem is that `f_nmtonname` is not defined in `statvfs` on Darwin.  There seems to be some code referencing choosing between `statfs` and `statvfs`.  It may be incorrectly choosing `statvfs` instead of `statfs`, because `statfs` does define `f_nmtonname`.

**Comment by cho-m** [comment]
> I think the problem is that `f_nmtonname` is not defined in `statvfs` on Darwin. There seems to be some code referencing choosing between `statfs` and `statvfs`. It may be incorrectly choosing `statvfs` instead of `statfs`, because `statfs` does define `f_nmtonname`.

Yeah, build script automatically picks `statvfs` as long as header exists https://gitlab.archlinux.org/pacman/pacman/-/blob/v6.0.1/meson.build#L178-179
```meson
foreach member : [
    ['struct stat', 'st_blksize', '''#include <sys/stat.h>'''],
    ['struct statvfs', 'f_flag', '''#include <sys/statvfs.h>'''],
    ['struct statfs', 'f_flags', '''#include <sys/param.h>
                                    #include <sys/mount.h>'''],
  ]
  have = cc.has_member(member[0], member[1], prefix : member[2])
  conf.set('HAVE_' + '_'.join([member[0], member[1]]).underscorify().to_upper(), have)
endforeach

...

if conf.has('HAVE_STRUCT_STATVFS_F_FLAG')
  conf.set('FSSTATSTYPE', 'struct statvfs')
```

One work around is to also check that not running `darwin`.

**Comment by danielnachun** [comment]
Thanks for that last tip @cho-m - I patched the `meson.build` to check make sure `statvfs` is not used if on Darwin and it now works.

**Comment by danielnachun** [comment]
I had to tweak the patch to fix an undefined macro on Catalina but everything works now.  I will definitely upstream these changes to the pacman upstream repo as they should help `makepkg` work better on macOS.

File: Formula/makepkg.rb [hunk 1]
```diff
@@ -31,23 +29,77 @@ class Makepkg < Formula
   uses_from_macos "m4" => :build
   uses_from_macos "libxslt"
 
+  on_macos do
+    depends_on "coreutils" => :test # for md5sum
+  end
+
+  patch :DATA
```
Comment (carlocab) [inline Formula/makepkg.rb:36]: This patch needs to be upstreamed.

File: Formula/makepkg.rb [hunk 2]
```diff
@@ -14,14 +14,12 @@ class Makepkg < Formula
     sha256 high_sierra: "c8f2f6999669c56b5e40e2608ad1e0adfe2c8eb73f8cef959a229856d21da6ed"
   end
 
-  disable! date: "2022-03-28", because: "depends on fakeroot which does not build"
-
   depends_on "asciidoc" => :build
```
Comment (carlocab) [inline Formula/makepkg.rb:17]: Is this still needed if we're not building docs?

File: Formula/makepkg.rb [hunk 3]
```diff
@@ -14,14 +14,12 @@ class Makepkg < Formula
...
-  depends_on "automake" => :build
   depends_on "docbook-xsl" => :build
-  depends_on "libtool" => :build
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
   depends_on "pkg-config" => :build
+  depends_on "python@3.10" => :build
   depends_on "bash"
   depends_on "fakeroot"
   depends_on "gettext"
```
Comment (carlocab) [inline Formula/makepkg.rb:25]: Is this needed on macOS?

File: Formula/makepkg.rb (hunk 1 above)
Comment (danielnachun) [reply Formula/makepkg.rb:36]: Definitely - I will add a link once I've submitted it to their Gitlab.

File: Formula/makepkg.rb (hunk 2 above)
Comment (danielnachun) [reply Formula/makepkg.rb:17]: Probably not - I will check this before the next push.

Comment (danielnachun) [reply Formula/makepkg.rb:17]: Also, we might be able to drop `docbook-xsl` here as well. I will update this whatever can be dropped.

File: Formula/makepkg.rb (hunk 3 above)
Comment (danielnachun) [reply Formula/makepkg.rb:25]: I just tried with `uses_from_macos "python" => :build` instead and it works, so it does not appear to need a specific version of Python 3 to build on macOS.  I will change this in the next push.

Comment (carlocab) [reply Formula/makepkg.rb:25]: I was actually referring to gettext, since we pass `i18n=false`.

Comment (danielnachun) [reply Formula/makepkg.rb]: I dropped the `gettext` dependency on macOS and it worked fine locally, so it is now a Linux-only dependency.

File: Formula/makepkg.rb (hunk 1 above)
Comment (danielnachun) [reply Formula/makepkg.rb:39]: I have submitted the patch to the mailing list and added a link to the mailing list archive.

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**